p2p/conn/secret_connection.go

```diff
@@ -334,11 +334,7 @@ func shareAuthSignature(sc *SecretConnection, pubKey crypto.PubKey, signature []
 
 // increment nonce big-endian by 1 with wraparound.
 func incrNonce(nonce *[aeadNonceSize]byte) {
-	for i := aeadNonceSize - 1; 0 <= i; i-- {
-		nonce[i]++
-		// if this byte wrapped around to zero, we need to increment the next byte
-		if nonce[i] != 0 {
-			return
-		}
-	}
+	counter := binary.LittleEndian.Uint64(nonce[4:])
```

Comment:
what about the first 4 bytes?

Reply:
We do not use them / they will be all zero. We use chacha20poly1305 which expects a [12 byte nonce] though.

[12 byte nonce]:  https://github.com/golang/crypto/blob/614d502a4dac94afa3a6ce146bd1736da82514c6/chacha20poly1305/chacha20poly1305.go#L24

Reply:
It's kind of weird, but only using the last 8 bytes has become the standard for cha cha nonces in stream mode. (This is what is done in noise)

This will be sufficient for our purposes tho, since 2^64 messages will be more than well have. (We can't use random nonces, since that only gave a 2^48 bound, which is in the reachable zone)